### PR TITLE
feature: validation for restricted ips and js domains

### PR DESCRIPTION
**Pull request by ch-rfishwick**
Added validation for restricted ip addresses and javascript domains on addKey and updateKey

### CODE REVIEW

File: server/lib/validation/form_validators/ManageApplication.js [hunk 1]
```diff
@@ -78,6 +78,8 @@ class ManageApplication extends Validator {
     logger.info('Validating key payload data');
     this.errors = {};
     this._formatIncomingPayload(payload);
+    let restrictedIps = payload.restrictedIps.split(',');
+    let javaScriptDomains =  payload.javaScriptDomains.split(',');
```
Comment (CH-JonathanBlake) [inline server/lib/validation/form_validators/ManageApplication.js:82]: Can we skip these tests if they are null? I would expect this to fail if no IP or Domains are provided.
https://stackoverflow.com/questions/9843394/javascript-cannot-call-method-split-of-null

**Review by CH-JonathanBlake — CHANGES_REQUESTED** [review]

File: server/lib/validation/form_validators/ManageApplication.js (hunk 1 above)
Comment (ch-rfishwick) [reply server/lib/validation/form_validators/ManageApplication.js]: I have moved them into a conditional block

File: server/lib/validation/form_validators/ManageApplication.js [hunk 2]
```diff
@@ -94,17 +92,23 @@ class ManageApplication extends Validator {
         this.errors.description = errorManifest.description.invalid;
       }
     }
-    for (const restrictedIp of restrictedIps){
-      if (restrictedIp.length > 0) {
-        if (!this.isValidIp(restrictedIp)) {
-          this.errors.restrictedIp = errorManifest.restrictedIp.invalid;
+    if (payload.restrictedIps.length > 0) {
```
Comment (CH-JonathanBlake) [inline server/lib/validation/form_validators/ManageApplication.js:95]: I'm surprised this handles nulls, I would have thought it still throws a null pointer when accessing length it payload doesn't have a value for.
Could we change that to `if !payload.restrictedIps { `
https://codereview.stackexchange.com/questions/5572/string-isnullorempty-in-javascript

**Review by CH-JonathanBlake — CHANGES_REQUESTED** [review]

File: server/lib/validation/form_validators/ManageApplication.js (hunk 2 above)
Comment (ch-rfishwick) [reply server/lib/validation/form_validators/ManageApplication.js:95]: won't that send it into the block if restrictedIps is empty?

Comment (CH-JonathanBlake) [reply server/lib/validation/form_validators/ManageApplication.js:95]: Oh yeah, so drop the `!` would fix that right? My mistake.

Comment (CH-JonathanBlake) [reply server/lib/validation/form_validators/ManageApplication.js:95]: or `if payload.restrictedIps && payload.restrictedIps.length > 0 {`

Comment (ch-rfishwick) [reply server/lib/validation/form_validators/ManageApplication.js]: Ok, changed

**Review by CH-JonathanBlake — APPROVED** [review]

**Review by ICraig-CH — APPROVED** [review]

**Pull request merged**